react-native-example/App.js

```diff
@@ -1,40 +1,56 @@
 import React from 'react';
 import { StyleSheet, Text, View } from 'react-native';
-import { Tracer, ExplicitContext, BatchRecorder, jsonEncoder } from 'zipkin';
+import { Tracer, ExplicitContext, BatchRecorder, jsonEncoder, Annotation } from 'zipkin';
 import wrapFetch from 'zipkin-instrumentation-fetch';
 import { HttpLogger } from 'zipkin-transport-http';
 
 const { JSON_V2 } = jsonEncoder;
 
 const zipkinBaseUrl = 'http://localhost:9411';
+const tracer = new Tracer({
+  ctxImpl: new ExplicitContext(),
+  recorder: new BatchRecorder({
+    logger: new HttpLogger({
+      endpoint: `${zipkinBaseUrl}/api/v2/spans`,
+      jsonEncoder: JSON_V2,
+      fetch,
+    }),
+  }),
+});
 
 export default class App extends React.Component {
   state = {
     person: 'Not loaded',
     fetch: wrapFetch(fetch, {
-      tracer: new Tracer({
-        ctxImpl: new ExplicitContext(),
-        recorder: new BatchRecorder({
-          logger: new HttpLogger({
-            endpoint: `${zipkinBaseUrl}/api/v2/spans`,
-            jsonEncoder: JSON_V2,
-            fetch,
-          }),
-        }),
-      }),
-      serviceName: 'react-native',
+      tracer,
+      serviceName: 'fetch-client',
       remoteServiceName: 'starwars',
     }),
   };
 
   componentWillMount() {
-    const randomNumber = Math.floor(Math.random() * 10) + 1;
-    this.state.fetch("https://swapi.co/api/people/" + randomNumber)
-      .then(response => response.json())
-      .then(person => this.setState({
-        person: person.name,
-      })
-      ).catch(e => console.error(e));
+    const id = tracer.createRootId();
+
+    tracer.scoped(() => {
+      tracer.setId(id);
+      tracer.recordAnnotation(new Annotation.ServerRecv());
```

Comment:
ps for "local spans" we tend to use client annotations. You might want to add a TODO or even better implement this! https://github.com/openzipkin/zipkin-js/issues/104